### ISSUE TITLE
feature request: support raw queries

### ISSUE THREAD

**Issue by Virgiel** [LouisGariepy/cornucopia#34]
I would like another quantifier for [raw queries](https://docs.rs/tokio-postgres/latest/tokio_postgres/struct.Client.html#method.query_raw). It allow me to perform calculation whiteout allocating a Vec or mapping rows before allocating.

**Comment by LouisGariepy** [comment]
Hi @Virgiel, thanks for opening this issue.

Just so we're on the same page, I'd like for you to confirm if this is what you're asking for :

Let's say we use `#` as a raw return identifier. Then this query
```sql
--! example (hello, world) #
/* some fancy SQL */
```
will generate something like this function
```rust
fn example(hello: &str, world: &i32) -> Result<tokio_postgres::RowStream, Error> {
/* Omitted for brevity */
}
```
is that right?

**Comment by LouisGariepy** [comment]
Implemented in branch https://github.com/LouisGariepy/cornucopia/tree/34-feature-request-support-raw-queries, waiting for more input.

**Comment by Virgiel** [comment]
To reuse the author example in the README, I was thinking of something like that:

``` sql
--! authors() #
SELECT * FROM Author;
```

```rust
 pub async fn authors<T: GenericClient>(
        client: &T,
    ) -> Result<impl Stream<Item = Result<(i32, String, String), Error>>, Error> {
        let stmt = client.prepare("SELECT * FROM Author;").await?;
        let res = client.query_raw(&stmt, &[]).await?;
        let stream = res.map(|res| {
            res.map(|res| {
                let return_value_0: i32 = res.get(0);
                let return_value_1: String = res.get(1);
                let return_value_2: String = res.get(2);
                (return_value_0, return_value_1, return_value_2)
            })
        });
        Ok(stream.into_stream())
    }
```

**Comment by Virgiel** [comment]
By the way, you could use query_raw and map when returning a Vec to prevent allocating twice.

**Comment by Virgiel** [comment]
When using `&[]` as empty params, rustc willl ask for type annotations. You can use `std::iter::empty::<i32>()` instead.

**Comment by LouisGariepy** [comment]
@Virgiel I am only a beginner with regards to lower-level programming, but I'm thrilled to improve the allocation performance of the generated code. I've opened a PR and I'll try to merge these improvements as soon as possible.

Thanks for your interest in Cornucopia, and more specifically for the awesome suggestions.

**Comment by LouisGariepy** [comment]
@Virgiel Just merged both of your suggestions in v0.7. Feel free to reopen or open a new issue if any more improvements are needed :smile: .

Thanks again!
